### ISSUE TITLE
Muscle wrapping seems to fail during run or during initialization for ellipsoid

### ISSUE THREAD

**Issue by joris997** [opensim-org/opensim-core#2883]
Hello all,
(OpenSim 4.1, Ubuntu 18.04, C/C++)

I've been trying to realize simple wrapping with objects different than cylinders. It seems that muscle wrapping fails (for the cases I studied) for ellipsoids. I've created a simple model consisting of two vertically moving equal masses, connected by a muscle which wraps over an object as shown below, below the two masses are two (invisible) physical springs which prevent the masses from jumping too far up during muscle activation;

Code for cylinder wrapping:
```
auto wrappingFrame = new PhysicalOffsetFrame("wrappingFrame", model.getGround(),
                                                 SimTK::Transform(Vec3(0, h, 0)));
    // Add the wrapping surface
    auto wrapSurface = new WrapCylinder();
    wrapSurface->setAllPropertiesUseDefault(true);
    wrapSurface->set_radius(r);
    wrapSurface->set_length(1);
    wrapSurface->set_xyz_body_rotation(Vec3(0));
    wrapSurface->set_quadrant("+y");
```

![Peek_cylinder](https://user-images.githubusercontent.com/56355937/94653558-0db65780-02fc-11eb-9798-aa6cf80c5167.gif)

With some muscle activation, the masses move as expected during simulation for the cylinder. However, changing the wrapping surface to an ellipsoid according to;

```
auto wrappingFrame = new PhysicalOffsetFrame("wrappingFrame", model.getGround(),
                                      SimTK::Transform(Vec3(0, h, 0)));
    // Add the wrapping surface
    auto wrapSurface = new WrapEllipsoid();
    wrapSurface->setAllPropertiesUseDefault(true);
    wrapSurface->set_dimensions(Vec3(r,r,1));
    wrapSurface->set_xyz_body_rotation(Vec3(0));
    wrapSurface->set_quadrant("+y");
```

May result in two situations depending on the height 'h' of the cylinder;

1. For a small height (0.5 m in this instance), the wrapping is initially correct but fails after a short time during simulation (notice how 'a bit of wrapping' seems to take place after the masses+muscle have bounced back above the ellipsoid);
![Peek_wrapping](https://user-images.githubusercontent.com/56355937/94652545-7d2b4780-02fa-11eb-9837-b80cc28cb338.gif)

2. For a larger height (1.0 m in this instance), the wrapping does not even initialize;
![Peek_no_wrapping](https://user-images.githubusercontent.com/56355937/94652564-83212880-02fa-11eb-86e4-dcf3fe1ac833.gif)

What I tried:

- Different Ellipsoid dimensions, rotations
- Different Ellipsoid displacement (not having the muscle straight over the center x-y plane)
- Changing the wrapping quadrant (given that the cylinder works as expected, I suspect that this is not the issue)

The code I used can be found [here](https://github.com/joris997/OpenSim_RA/tree/master/Analytical_wrapping) but some changes in the hopper example results in the same behavior. In my linked code, the cylinder height parameters is called `tc.CYLINDER_HEIGHT` instead of `h` which is from the passed argument `tc`. This is a struct for which default values can be changed in testCase.h)

Given that this issue can be replicated by changing a minimal amount of code from a situation with a cylinder that works like a charm, leads me to believe that this is not an implementation issue on my part and therefore fits as an issue to the opensim-core source code repository.

**Comment by adamkewley** [comment]
For anyone else, here's a standalone version of the model that only depends on `OpenSim.h`:

https://gist.github.com/adamkewley/2d7447a536000b2adc67a37dae205201

**Comment by jenhicks** [comment]
@aymanhab Would any of the recent work you've been doing related to wrapping be relevant to this issue?

**Comment by aymanhab** [comment]
@jenhicks My recent work focused on performance only and I wouldn't expect it to change anything in Ellipsoid wrapping. If anything, the inspection of the code and the heuristics built into it makes this behavior rather unsurprising :confused:

**Comment by adamkewley** [comment]
Yeah, sorry about not grinding this one: it's been kicked back because of everything else going on. I'm in the same boat as @aymanhab in that a quick inspection of the code made it clear that these kinds of bugs would probably pop up

**Comment by adamkewley** [comment]
@pepbos - this old issue is potentially an interesting edge-case if you are thinking of changing the cylinder wrapping code. @joris997 found it when he was investigating modelling issues.

It was a while ago, but the Chineese proverb "The faintest ink is better than the strongest memory" comes to mind
